docs/user/setting_up_your_development_environment.md

```diff
@@ -0,0 +1,101 @@
+# Setting up your development environment
+
+## Overview
+
+Garage is a Python package which can be installed in most Python 3.6+
+environments using standard commands, i.e.
+
+``pip install --user garage``
+
+However, to ensure that dependencies get resolved properly, we suggest
+using a Python environment manager that supports dependency resolution,
+such as [pipenv](https://docs.pipenv.org/en/latest/),
+[conda](https://docs.conda.io/en/latest/), or
+[poetry](https://poetry.eustace.io/). We test against pipenv and conda.
+
+Garage is also tested in [virtualenv](https://virtualenv.pypa.io/en/latest/), but
+virtualenv occasionally has difficulty resolving dependency conflicts, so use with
+care.
+
+_Note: Starting with our 2020.06 release, we only support Python 3.6+._
+
+## Installing Dependencies
+
+Generally speaking, system dependencies of garage are minimal, and likely already
+installed. However, many of the environments garage is used with have additional
+dependencies, and we provide "setup scripts" for installing those dependencies
+and working around known problems on common platforms.
+
+If you can already use the environments you need, you can skip this section.
+
+A MuJoCo key is required for full functionality. You can get one
+[here](https://www.roboti.us/license.html).
+
+### Running the installation scripts
```

Comment:
you should note that, while optional for regular usage, the dev install has *all* features so this is virtuall required